pkg/receive/handler.go

```diff
@@ -196,6 +243,27 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 		h.replicationFactor.Set(1)
 	}
 
+	h.ActiveSeriesLimit.configuredTenantLimit.Set(float64(o.MaxPerTenantLimit))
+	h.ActiveSeriesLimit.tenantCurrentSeriesMap = map[string]float64{}
```

Comment:
Do we need a constructor function for `ActiveSeriesLimit`?

Also, have you considered having an interface and a `NopSeriesLimiter` that is injected in the handler when no limits are applied? This should avoid a lot of the partial state and explicit checks throughout the codebase.

Reply:
That's a good idea! Let me update the PR with something like this!